### PR TITLE
OSDOCS-7553: 1.4 Network Observability UI Enhancements

### PR DESCRIPTION
**Pull request by skrthomas**
Version(s):

Issue:

https://issues.redhat.com/browse/OSDOCS-7553
Link to docs preview:

https://63885--docspreview.netlify.app/openshift-enterprise/latest/networking/network_observability/observing-network-traffic#network-observability-quickfilternw-observe-network-traffic
QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by skrthomas** [comment]
@jotak can you have a look at the updates for the UI enhancements? I _think_ this covers everything we discussed in doc planing, but let me know if I'm missing anything.

File: modules/network-observability-quickfilter.adoc [hunk 1]
```diff
@@ -11,19 +11,22 @@ Query Options::
 You can use *Query Options* to optimize the search results, as listed below:
 
 ** *Log Type*: The available options *Conversation* and *Flows* provide the ability to query flows by log type, such as flow log, new conversation, completed conversation, and a heartbeat, which is a periodic record with updates for long conversations. A conversation is an aggregation of flows between the same peers.  
-** *Reporter Node*: Every flow can be reported from both source and destination nodes. For cluster ingress, the flow is reported from the destination node and for cluster egress, the flow is reported from the source node. You can select either *Source* or *Destination*. The option *Both* is disabled for the *Overview* and *Topology* view. The default selected value is *Destination*.
+** *Duplicated flows*: A flow might be reported from several interfaces, and from both source and destination nodes, making it appear in the data several times. By selecting this query option, you can choose to show duplicates. Duplicates are hidden by default. Use the *Direction* filter in the *Common* section in the to switch between ingress and egress traffic.
```
Comment (jotak) [inline modules/network-observability-quickfilter.adoc:14]: A word seems missing in the last sentence: `Use the Direction filter in the Common section in the (filters dropdown?) to switch between ingress and egress traffic`

File: modules/network-observability-quickfilter.adoc [hunk 2]
```diff
@@ -11,19 +11,22 @@ Query Options::
...
 ** *Limit*: The data limit for internal backend queries. Depending upon the matching and the filter settings, the number of traffic flow data is displayed within the specified limit.
 
 Quick filters::
 The default values in *Quick filters* drop-down menu are defined in the `FlowCollector` configuration. You can modify the options from console. 
 
 Advanced filters::
-You can set the advanced filters by providing the parameter to be filtered and its corresponding text value. The section *Common* in the parameter drop-down list filters the results that match either *Source* or *Destination*. To enable or disable the applied filter, you can click on the applied filter listed below the filter options.
+You can set the advanced filters by providing the parameter to be filtered and its corresponding text value. The *Common* section of filters the results based on the selection, such as *Protocol*, *Direction*, *Network latency*, *Packet drop TCP state*, *Packet drop latent cause*, *DNS Latency*, *DNS Id*, *DNS response code*. To enable or disable the applied filter, you can click on the applied filter listed below the filter options.
```
Comment (jotak) [inline modules/network-observability-quickfilter.adoc:22]: `The *Common* section of filters the results based on ...` => remove `of` ?

Comment (jotak) [reply modules/network-observability-quickfilter.adoc:22]: I'm not sure it's necessary to list the "Common" filters now, actually. We don't list all available filters under "Source" and "Destination", the ones under "Common" aren't more important.

File: modules/network-observability-quickfilter.adoc (hunk 1 above)
Comment (skrthomas) [reply modules/network-observability-quickfilter.adoc:14]: Yep you're right I didn't finish this phrase. I think I was experimenting with how to phrase filter dropdowns and went below to advanced filters and forgot to come back.

[the remark this answers is not in the document]
Comment (skrthomas) [reply modules/network-observability-quickfilter.adoc:14]: @jotak That sounds good to me. One question about the details you mention about the `Interface` and `Direction`  fields. Would a customer use the Duplicated flows query option and then also filter further with the `Direction` filter? Or is this a different direction field you're talking about? Also, where is the `Interface` field and how would a customer use it?

**Comment by skrthomas** [comment]
@jotak @memodi @jpinsonneau hey guys, can you PTAL at the documentation changes for the UI updates? Let me know if anything needs addressing. TY :)

File: modules/network-observability-quickfilter.adoc [hunk 3]
```diff
@@ -11,19 +11,22 @@ Query Options::
...
+You can set the advanced filters, *Common*, *Source*, or *Destination*, by selecting the parameter to be filtered from the dropdown list. The flow data is filtered based on the selection. To enable or disable the applied filter, you can click on the applied filter listed below the filter options.
 
 [NOTE]
 ====
 To understand the rules of specifying the text value, click *Learn More*.
 ====
 
-You can click *Reset default filter* to remove the existing filters, and apply the filter defined in `FlowCollector` configuration.
\ No newline at end of file
+You can toggle between image:arrow-up-long-solid.png[,10] *One way* and image:arrow-up-long-solid.png[,10] image:arrow-down-long-solid.png[,10] *Back and forth* filtering. The image:arrow-up-long-solid.png[,10] *One way* filter shows *Source* or *Destination* traffic according to your filter selections. The image:arrow-up-long-solid.png[,10] image:arrow-down-long-solid.png[,10] *Back and forth* filter includes return traffic to your existing *Source* or *Destination* filters.
```
Comment (jpinsonneau) [inline modules/network-observability-quickfilter.adoc:29]: Is it worth mentionning that you can **swap** `Source` and `Destination` ? 
That's useful using one way.

image
image

Comment (skrthomas) [reply modules/network-observability-quickfilter.adoc:29]: I agree, we should mention it and the utility of using it when filtering one way.

**Comment by skrthomas** [comment]
@jpinsonneau @jotak does everything look OK now from your view?

**Review by jpinsonneau — APPROVED** [review]

File: modules/network-observability-quickfilter.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (jotak) [reply modules/network-observability-quickfilter.adoc]: > I was wondering if there was a gap now that the Common filter doesn't match both Source and Destination.

There shouldn't be a gap, users can still see in/out flows for a given resource by check the new "back and forth" option

> Would a customer use the Duplicated flows query option and then also filter further with the Direction filter?

Yes it's possible to show duplicates & filter with Direction and/or show the Direction column to get a deeper insight of how & when flows are captured. Though not sure we need to go that deep in the doc...

> Also, where is the Interface field and how would a customer use it?

Interface is available both as a filter and a displayable column

**Review by jotak — APPROVED** [review]
looks great, thanks!

File: modules/network-observability-quickfilter.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (skrthomas) [reply modules/network-observability-quickfilter.adoc]: > There shouldn't be a gap, users can still see in/out flows for a given resource by check the new "back and forth" option
AH, okay. I want to clarify that I didn't mean a gap in user experience I meant a gap in content explanation for how to see these flows. Sounds like we've got it covered though so I think its all good. I always worry I'm missing small little details for things 😅 

Thanks for the info to both those other questions.

**Review by ShaunaDiaz — APPROVED** [review]
Looks great to me! Just one question.

**Review by memodi — APPROVED** [review]
/label qe-approved

File: modules/network-observability-quickfilter.adoc [hunk 4]
```diff
@@ -11,19 +11,22 @@ Query Options::
...
-** *Reporter Node*: Every flow can be reported from both source and destination nodes. For cluster ingress, the flow is reported from the destination node and for cluster egress, the flow is reported from the source node. You can select either *Source* or *Destination*. The option *Both* is disabled for the *Overview* and *Topology* view. The default selected value is *Destination*.
+** *Duplicated flows*: A flow might be reported from several interfaces, and from both source and destination nodes, making it appear in the data several times. By selecting this query option, you can choose to show duplicated flows. Duplicated flows have the same sources and destinations, including ports, and also have the same protocols, with the exception of `Interface` and `Direction` fields. Duplicates are hidden by default. Use the *Direction* filter in the *Common* section of the filters dropdown to switch between ingress and egress traffic.
```
[the remark this answers is not in the document]
Comment (skrthomas) [reply modules/network-observability-quickfilter.adoc:14]: AH, that's a really good catch. You're right, dropdown list is much better. TY!

[the remark this answers is not in the document]
Comment (skrthomas) [reply modules/network-observability-quickfilter.adoc:14]: In this case, I don't think dropdown needs to be bold.

**Pull request merged**